codice/ddf
# [2.13.x] Changed PKI certificate validation

#### What does this PR do?
PKI authentication certificates will now be validated against the truststore.

#### Who is reviewing it? 
@bakejeyner @SmithJosh 

#### Select relevant component teams: 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@garrettfreibott
@stustison 

#### How should this be tested?
- Create a CA (`rootCA`)
- Create a certificate using the `rootCA` (`cert`)
- Add the `rootCA` to DDF's **keystore** (**NOT the truststore**)
- Use the `cert` to login to DDF -> this should fail

- **Remove** the `rootCA` from the DDF's **keystore**
- **Add** the `rootCA` to DDF's **truststore**
- Use the `cert` to login to DDF -> this should pass

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
